controllers/notification/mailgun_notification_delivery_service_test.go

```diff
@@ -186,11 +196,57 @@ func TestMailgunNotificationDeliveryService(t *testing.T) {
 				Template:  "replyto",
 				Context:   notCtx,
 			},
-		})
+		}, notificationtemplates.SandboxNotificationTemplateSetName)
 
 		// then
 		require.NoError(t, err)
 		require.Equal(t, "a message sent to info@foo.com", formValues.Get("html"))
 
 	})
+
+	t.Run("test mailgun notification delivery empty subject template", func(t *testing.T) {
+		// when
+		mgds := NewMailgunNotificationDeliveryService(config, templateLoader, mockServerOption)
+		err := mgds.Send(&toolchainv1alpha1.Notification{
+			Spec: toolchainv1alpha1.NotificationSpec{
+				Template: "empty_subject",
+				Context:  notCtx,
+			},
+		}, notificationtemplates.SandboxNotificationTemplateSetName)
+
+		// then
+		require.Error(t, err)
+		require.Equal(t, "no subject or body specified for notification", err.Error())
+	})
+
+	t.Run("test mailgun notification delivery empty content template", func(t *testing.T) {
+		// when
+		mgds := NewMailgunNotificationDeliveryService(config, templateLoader, mockServerOption)
+		err := mgds.Send(&toolchainv1alpha1.Notification{
+			Spec: toolchainv1alpha1.NotificationSpec{
+				Template: "empty_content",
+				Context:  notCtx,
+			},
+		}, notificationtemplates.SandboxNotificationTemplateSetName)
+
+		// then
+		require.Error(t, err)
+		require.Equal(t, "no subject or body specified for notification", err.Error())
+	})
+
+	t.Run("test mailgun notification delivery send for appstudio", func(t *testing.T) {
+		// when
+		mgds := NewMailgunNotificationDeliveryService(config, templateLoader, mockServerOption)
+		err := mgds.Send(&toolchainv1alpha1.Notification{
+			Spec: toolchainv1alpha1.NotificationSpec{
+				Recipient: "foo@bar.com",
+				Subject:   "test",
+				Content:   "abc",
+				Context:   notCtx,
+			},
+		}, notificationtemplates.AppstudioNotificationTemplateSetName)
```

Comment:
this whole `TestMailgunNotificationDeliveryService` test can be changed to iterate over the set names:
https://gist.github.com/MatousJobanek/1496f2538a4ea71280f363594dfc1123

Reply:
done in [29be8b4](https://github.com/codeready-toolchain/host-operator/pull/754/commits/29be8b41347898468009c261876303dde8b37305)